src/components/CookieClicker.vue

```diff
@@ -0,0 +1,411 @@
+      }
+    },
+    mounted() {
+      this.loadGame();
+      this.startGameLoop();
+    },
+    beforeDestroy() {
+      clearInterval(this.gameLoop);
+    },
+    methods: {
+      increment() {
+        this.cookies += 1;
+        this.totalClicks++;
+        this.checkAchievements();
+      },
+      getUpgradeStyle(index) {
+    const positions = [
+      { left: '15%', top: '20%' },
+      { left: '35%', top: '30%' },
+      { left: '75%', top: '25%' }
+    ];
+    return {
+      ...positions[index],
+      transform: `scale(${0.8 + this.availableUpgrades[index].owned * 0.1})`
+    };
+  },
+
+  
+  purchaseUpgrade(index) {
+    const upgrade = this.availableUpgrades[index];
+    if (this.cookies >= upgrade.cost) {
+      this.cookies -= upgrade.cost;
+      upgrade.owned += 1; 
+      this.cookiesPerSecond += upgrade.value;
+      upgrade.cost = Math.floor(upgrade.cost * 1.15);
+      this.saveGame(); 
+      
+
+      this.$forceUpdate(); 
+      
+      console.log(`Purchased ${upgrade.name}`, upgrade);
+    }
+  },
+      },
+      checkAchievements() {
+
+      },
+      startGameLoop() {
+        this.gameLoop = setInterval(() => {
+          this.cookies += this.cookiesPerSecond / 10;
+          this.saveGame();
+        }, 100);
+      },
+      formatNumber(num) {
+        return num.toFixed(0).replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+      },
+      saveGame() {
+        const gameData = {
+          cookies: this.cookies,
+          upgrades: this.availableUpgrades,
+          cps: this.cookiesPerSecond
+        };
+        localStorage.setItem('cookieClickerSave', JSON.stringify(gameData));
+      },
+      loadGame() {
+        const savedData = localStorage.getItem('cookieClickerSave');
+        if (savedData) {
+          const gameData = JSON.parse(savedData);
+          this.cookies = gameData.cookies || 0;
+          this.availableUpgrades = gameData.upgrades || this.availableUpgrades;
+          this.cookiesPerSecond = gameData.cps || 0;
+        }
+      },
+      handleImageError(type) {
+        const emojis = {
+          cursor: '🖱️',
+          grandma: '👵',
+          farm: '🚜'
+        };
+        
+        const svg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100">
+          <rect width="100" height="100" fill="#f0f0f0"/>
+          <text x="50" y="60" font-size="60" text-anchor="middle">${emojis[type]}</text>
+        </svg>`;
+        
+        const dataUri = `data:image/svg+xml;utf8,${encodeURIComponent(svg)}`;
+        
+        if (type === 'cursor') this.cursorImage = dataUri;
+        if (type === 'grandma') this.grandmaImage = dataUri;
+        if (type === 'farm') this.farmImage = dataUri;
+      },
+      handleCookieError() {
+        this.cookieImage = 'data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100"><text x="50" y="60" font-size="60" text-anchor="middle">🍪</text></svg>';
+      }
+    }
+  
+  </script>
+  
+  <style scoped>
```

Comment:
Styling is fine but most of it is pretty similar to bulma so either customize that or?